addons/google_account/controllers/main.py

```diff
@@ -1,32 +1,30 @@
+# -*- coding: utf-8 -*-
+
 import json
-import urllib
-import openerp
-from openerp import http
-from openerp.http import request
-import openerp.addons.web.controllers.main as webmain
-from openerp.addons.web.http import SessionExpiredException
-from werkzeug.exceptions import BadRequest
 import werkzeug.utils
 
-class google_auth(http.Controller):
-    
+from odoo import http
+from odoo.http import request
+from odoo.modules.registry import RegistryManager
+
+
+class GoogleAuth(http.Controller):
+
     @http.route('/google_account/authentication', type='http', auth="none")
     def oauth2callback(self, **kw):
         """ This route/function is called by Google when user Accept/Refuse the consent of Google """
-        
+
         state = json.loads(kw['state'])
         dbname = state.get('d')
         service = state.get('s')
         url_return = state.get('f')
-        
-        registry = openerp.modules.registry.RegistryManager.get(dbname)
+
+        registry = RegistryManager.get(dbname)
```

Comment:
How we can create new Environment instead of RegistryManager
Can you explain in detail....? 

https://github.com/odoo/odoo/blob/master/openerp/api.py#L785
Here also api.Environment used RegistryManager